src/server/svr_jobfunc.c

```diff
@@ -4601,21 +4601,20 @@ start_end_dur_wall(void *pobj, int objtype)
 			swcode += 8;			/*have walltime*/
 		else if (!(prsc = add_resource_entry(pattr, rscdef)))
 			return (-1);
-	}
-	else {
-		if (presv->ri_alter_flags & RESV_DURATION_MODIFIED) 
-				swcode += 4;
-		if (presv->ri_alter_flags & RESV_END_TIME_MODIFIED)
-				swcode += 2; /*calcualte start time*/
-		if (presv->ri_alter_flags & RESV_START_TIME_MODIFIED)
-				swcode += 1; /*calculate end time*/;
+	} else {
+		swcode = 3;
+		if (presv->ri_alter_flags & RESV_DURATION_MODIFIED) {
+			swcode +=1;
+			if (presv->ri_alter_flags & RESV_END_TIME_MODIFIED)
+					swcode += 2; /*calcualte start time*/
+			if (presv->ri_alter_flags & RESV_START_TIME_MODIFIED)
+					swcode += 1; /*calculate end time*/;
+		}
```

Comment:
If duration is the only thing modified, this would make swcode 4, which is wrong.
```
		if (presv->ri_alter_flags & RESV_START_TIME_MODIFIED)
				swcode += 1; /* use start time*/
		if (presv->ri_alter_flags & RESV_END_TIME_MODIFIED)
				swcode += 2; /* use end time*/
		if (presv->ri_alter_flags & RESV_DURATION_MODIFIED)
				swcode += 4; /* use duration */
		if (presv->ri_alter_flags == RESV_START_TIME_MODIFIED || presv->ri_alter_flags == RESV_END_TIME_MODIFIED)
				swcode = 3; /* only start or only end changed, use start and end to calculate duration */
```

Reply:
Alternatively, you can do what you did before, which was add the switch cases for 1 and 2, and you just have to put a 
```C
        case 1:
        case 2:
        /* states only valid for altered reservations */ 
        if (pstate != RESV_BEING_ALTERED)
           return -1;
        /* fall through */
        case 3:
```

Reply:
It can be 4. The case for 4 and 5 is same. Also according to the current code it will never be 4. If we change only duration we make sure to set RESV_START_TIME_MODIFIED flag